src/components/autocomplete/js/autocompleteDirective.js

```diff
@@ -0,0 +1,58 @@
+(function () {
+  'use strict';
+  angular
+      .module('material.components.autocomplete')
+      .directive('mdAutocomplete', MdAutocomplete);
+
+  function MdAutocomplete () {
+    return {
+      template: '\
+        <md-autocomplete-wrap role="listbox">\
+          <input type="text"\
+              ng-model="searchText"\
+              ng-keydown="$mdAutocompleteCtrl.keydown($event)"\
+              placeholder="{{placeholder}}"\
+              aria-label="{{placeholder}}"\
+              aria-autocomplete="list"\
+              aria-haspopup="true"\
+              aria-activedescendant=""\
+              aria-expanded="{{!$mdAutocompleteCtrl.hidden}}"\
+              ng-keydown="$mdAutocompleteCtrl.onKeydown($event)"/>\
+          <button\
+              type="button"\
+              ng-if="searchText"\
+              ng-click="$mdAutocompleteCtrl.clear()">\
+              <span aria-hidden="true">X</span>\
+              <span class="visually-hidden">Clear</span>\
+              </button>\
+          <md-progress-linear ng-if="$mdAutocompleteCtrl.loading" md-mode="indeterminate"></md-progress-linear>\
+        </md-autocomplete-wrap>\
+        <ul role="presentation">\
+          <li ng-repeat="(index, item) in $mdAutocompleteCtrl.matches"\
```

Comment:
This is more personal preference than anything, but I like to avoid using variables that I don't explicitly name.  It's not super important in this case, but I still find it cleaner to specify my own variables.